models/articles.js

```diff
@@ -1,18 +1,41 @@
 const db = require("../db/connection");
 
 exports.selectArticles = () => {
-  return db
-    .query(
-      `
+  const findQuery = `
   SELECT articles.author, articles.title, articles.article_id,  articles.topic, articles.created_at, articles.votes, articles.article_img_url,
   COUNT (comments.article_id)::integer as comment_count
   FROM articles
   LEFT JOIN comments ON articles.article_id = comments.article_id
   GROUP BY articles.article_id
   ORDER BY created_at DESC;
-`
-    )
-    .then((result) => {
-      return result.rows;
+`;
+
+  return db.query(findQuery).then((result) => {
+    return result.rows;
+  });
+};
+
+exports.selectArticleById = (id) => {
+  if (!/^\d+$/.test(id)) {
+    return Promise.reject({
```

Comment:
Whilst this is sufficient to pass the test you have done for a 400. Postgres will return an error if the type of the id is invalid. This would throw an error that would be passed to the catch block of your controller that you could handle in your error handling middleware